core/include/gnuradio-4.0/Block.hpp

```diff
@@ -1483,15 +1482,15 @@ class Block : public lifecycle::StateMachine<Derived> {
             // See https://github.com/fair-acc/gnuradio4/issues/444
             for_each_reader_span(
                 [&processedIn](auto& in) {
-                    if (in.isConsumeRequested() && in.isConnected) {
+                    if (in.isConsumeRequested() && in.isConnected && in.isSync) {
                         processedIn = std::min(processedIn, in.nRequestedSamplesToConsume());
                     }
                 },
                 inputSpans);
 
             for_each_writer_span(
                 [&processedOut](auto& out) {
-                    if (out.isPublishRequested() && out.isConnected) {
+                    if (out.isPublishRequested() && out.isConnected && out.isSync) {
```

Comment:
The current logic requires this behavior; otherwise, if the output async port has no samples, tags won’t be forwarded across all ports.

We still need to address certain sync-async port semantics, as highlighted in issues like [#369](https://github.com/fair-acc/gnuradio4/issues/369) and [#444](https://github.com/fair-acc/gnuradio4/issues/444).